builtin/mainmenu/dlg_create_world.lua

```diff
@@ -17,13 +17,120 @@
+local function check_flag(flags, flag)
+	return flags[flag] == true
+end
+
+-- Same as check_flag but returns a string
+local function strflag(flags, flag)
+	return (flags[flag] == true) and "true" or "false"
+end
+
+local cb_caverns = { "caverns", fgettext("Caverns"), "caverns",
+	fgettext("Very large caverns deep in the underground") }
+local tt_sea_rivers = fgettext("Sea level rivers")
+
+local flag_checkboxes = {
+	v5 = {
+		cb_caverns,
+	},
+	v7 = {
+		cb_caverns,
+		{ "ridges", fgettext("Rivers"), "ridges", tt_sea_rivers },
+		{ "mountains", fgettext("Mountains"), "mountains" },
+		{ "floatlands", fgettext("Floatlands (experimental)"), "floatlands",
+		fgettext("Floating landmasses in the sky") },
+	},
+	carpathian = {
+		cb_caverns,
+		{ "rivers", fgettext("Rivers"), "rivers", tt_sea_rivers },
+	},
+	valleys = {
+		{ "altitude-chill", fgettext("Altitude chill"), "altitude_chill",
+		fgettext("Reduces heat with altitude") },
+		{ "altitude-dry", fgettext("Altitude dry"), "altitude_dry",
+		fgettext("Reduces humidity with altitude") },
+		{ "humid-rivers", fgettext("Humid rivers"), "humid_rivers",
+		fgettext("Increases humidity around rivers") },
+		{ "vary-river-depth", fgettext("Vary river depth"), "vary_river_depth",
+		fgettext("Low humidity and high heat causes shallow or dry rivers") },
+	},
+	flat = {
+		{ "hills", fgettext("Hills"), "hills" },
+		{ "lakes", fgettext("Lakes"), "lakes" },
+	},
+	fractal = {
+		{ "terrain", fgettext("Terrain"), "terrain",
+		fgettext("Generate non-fractal terrain: Oceans and underground") },
+	},
+	v6 = {
+		{ "trees", fgettext("Trees and jungle grass"), "trees" },
+		{ "flat", fgettext("Flat terrain"), "flat" },
+		{ "mudflow", fgettext("Mud flow"), "mudflow",
+		fgettext("Terrain surface erosion") },
+		-- Biome settings are in mgv6_biomes below
+	},
+}
+
+local mgv6_biomes = {
+	{
+		fgettext("Temperate, Desert, Jungle, Tundra, Taiga"),
+		{jungles = true, snowbiomes = true}
+	},
+	{
+		fgettext("Temperate, Desert, Jungle"),
+		{jungles = true, snowbiomes = false}
+	},
+	{
+		fgettext("Temperate, Desert"),
+		{jungles = false, snowbiomes = false}
+	},
+}
+
 local function create_world_formspec(dialogdata)
 	local mapgens = core.get_mapgen_names()
 
 	local current_seed = core.settings:get("fixed_map_seed") or ""
 	local current_mg   = core.settings:get("mg_name")
 	local gameid = core.settings:get("menu_last_game")
 
+	local flags = {}
+	flags.main = flags_to_table(core.settings:get("mg_flags"))
+	flags.v5 = flags_to_table(core.settings:get("mgv5_spflags"))
+	flags.v6 = flags_to_table(core.settings:get("mgv6_spflags"))
+	flags.v7 = flags_to_table(core.settings:get("mgv7_spflags"))
+	flags.fractal = flags_to_table(core.settings:get("mgfractal_spflags"))
+	flags.carpathian = flags_to_table(core.settings:get("mgcarpathian_spflags"))
+	flags.valleys = flags_to_table(core.settings:get("mgvalleys_spflags"))
+	flags.flat = flags_to_table(core.settings:get("mgflat_spflags"))
```

Comment:
Maybe these should use:
https://github.com/minetest/minetest/blob/7c0e0c34ecb66e044f59cbab03a4c12fc7838e97/doc/lua_api.txt#L4532-L4542
?

Reply:
Apparently this does not work in main menu.